src/KafkaClient/Connections/ISslConfiguration.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Net.Security;
+
+namespace KafkaClient.Connections
+{
+    public interface ISslConfiguration
```

Comment:
There is no really good way to have a default implementation, because the client-side ssl setup must match with the server-side settings. For security reasons it's advised to check certificate properties, (like the owner/ussuer is matches what You expect etc.), and obviously there are no default values for these.
I've still created BasicSslConfiguration to have en example implementation.